src/controllers/keystore/keystore.ts

```diff
@@ -104,6 +105,12 @@ export class KeystoreController extends EventEmitter {
 
   #keystoreSeeds: KeystoreSeed[] = []
 
+  // when importing a seed, save it temporary here before deciding
+  // whether to place it in #keystoreSeeds or delete it
+  //
+  // this should be done only if there isn't a saved seed already
+  #tempSeed: KeystoreSeed | undefined = undefined
```

Comment:
Minor: I usually set the empty state to `null` when set on purpose. Undefined is the default (if you access the a prop of the object that doesn't exist). Explicitly setting an empty object property to `null` is generally better because it conveys intentionality.